src/modalities/checkpointing/checkpoint_saving_strategies.py

```diff
@@ -55,6 +85,15 @@ def get_checkpoint_instruction(
 
 class SaveEveryKStepsCheckpointingStrategy(CheckpointSavingStrategyIF):
     def __init__(self, k: int):
+        """
+        Initializes the CheckpointSavingStrategy object.
+
+        Args:
+            k (int): The value of k.
```

Comment:
This docstring basically does not say anything, that cannot be seen directly from looking at the code. It is redundant. 
It would be better to have a doc string that actually explains what we do here. 

For instance: 
"Initializes checkpointing strategy which saves a checkpoint every k steps."